### PR TITLE
refactor: Add IDs to elements of landing, login, register, and what's new screens

### PR DESCRIPTION
**Pull request by farhan-arshad-dev**
### Description
Add IDs to elements of landing, login, register, and what's new screens.

**Landing Screen:**
- The title under the App logo (Courses and programs from the world's best universities in your pocket)
- Text: (What do you want to learn?)
- Textfield (Search our 3000+ courses)
- Button: (Explore all courses)
- Button: Register
- Button: Sign in

**Login Screen:**

- Title: (Sign In)
- Text: (Welcome back! Please authorize to continue.)
- Email and Password (Labels and TextFields)
- Button: (Forgot password?)
- Button (Sign In)
- Text: (Or sign in with:)
- Buttons: (Sign in with Google, Facebook, Microsoft and Apple)
- On Forgot password page (Title, Text, Email label, Email field and Reset password button)

**Register Screen:**

- Title: (Sign up)
- Text: (Create new account.)
- All Input Fields with Labels and notes under the Fields (Full name, public name, email, password)
- Add ID and text: (Country or Region of Residence dropdown and its label)
- ID to all buttons (Create account, Register with facebook, Microsoft and Apple)
- Dropdowns: (Highest education and Gender)
- Dropdown values (Country, education and Gender)

**Whats new screen:**

- Title and Close Button
- Next, Previous & Done Buttons

### CODE REVIEW

File: auth/src/main/java/org/openedx/auth/presentation/signin/compose/SignInView.kt [hunk 1]
```diff
@@ -276,7 +291,9 @@ private fun SocialLoginView(
 ) {
     if (state.isGoogleAuthEnabled) {
         OpenEdXOutlinedButton(
-            modifier = buttonWidth.padding(top = 24.dp),
+            modifier = buttonWidth
+                .testTag("btn_google_auth")
```
Comment (HamzaIsrar12) [inline auth/src/main/java/org/openedx/auth/presentation/signin/compose/SignInView.kt:295]: Would providing the ID to the Text Field inside the Row container be beneficial for Bilal? Same goes for other Social Buttons as well.

File: auth/src/main/java/org/openedx/auth/presentation/logistration/LogistrationFragment.kt [hunk 2]
```diff
@@ -135,17 +143,22 @@ private fun LogistrationScreen(
...
-                        modifier = Modifier.padding(bottom = 10.dp),
+                        modifier = Modifier
+                            .testTag("txt_search_label")
+                            .padding(bottom = 10.dp),
                         style = MaterialTheme.appTypography.titleMedium,
                         text = stringResource(id = R.string.pre_auth_search_title),
                     )
                     SearchBar(
                         modifier = Modifier
+                            .testTag("tf_discovery_search")
```
Comment (HamzaIsrar12) [inline auth/src/main/java/org/openedx/auth/presentation/logistration/LogistrationFragment.kt:161]: Tf stands for? Text Field? 🤔

File: auth/src/main/java/org/openedx/auth/presentation/restore/RestorePasswordFragment.kt [hunk 3]
```diff
@@ -188,6 +197,7 @@ private fun RestorePasswordScreen(
 
             Surface(
                 modifier = Modifier
+                    .testTag("txt_forgot_password_description")
```
Comment (HamzaIsrar12) [inline auth/src/main/java/org/openedx/auth/presentation/restore/RestorePasswordFragment.kt:200]: I don't think we need one here. Plus the id is a duplicate of the actual description.

**Review by HamzaIsrar12 — CHANGES_REQUESTED** [review]
Great work overall. Minor nits only. 

_ps: I will coordinate with Bilal as well for verification._

File: auth/src/main/java/org/openedx/auth/presentation/signin/compose/SignInView.kt (hunk 1 above)
Comment (farhan-arshad-dev) [reply auth/src/main/java/org/openedx/auth/presentation/signin/compose/SignInView.kt:295]: yes, requested by the Automation team. assign the id to the Text.

File: auth/src/main/java/org/openedx/auth/presentation/logistration/LogistrationFragment.kt (hunk 2 above)
Comment (farhan-arshad-dev) [reply auth/src/main/java/org/openedx/auth/presentation/logistration/LogistrationFragment.kt:161]: yes.

File: auth/src/main/java/org/openedx/auth/presentation/restore/RestorePasswordFragment.kt (hunk 3 above)
Comment (farhan-arshad-dev) [reply auth/src/main/java/org/openedx/auth/presentation/restore/RestorePasswordFragment.kt:200]: good catch.

**Review by HamzaIsrar12 — APPROVED** [review]
LGTM, also checked it with Bilal Awan. 👍🏻

**Comment by bilalawan321** [comment]
Hey @farhan-arshad-dev 
I have verified that I can now see IDs on all the elements.

**Review by volodymyr-chekyrta — APPROVED** [review]

**Review by k1rill — APPROVED** [review]

**Pull request merged**